tiatoolbox/tools/tissuemask.py

```diff
@@ -0,0 +1,289 @@
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# This file contains code inspired by StainTools
+# [https://github.com/Peter554/StainTools] written by Peter Byfield.
+#
+# The Original Code is Copyright (C) 2020, TIALab, University of Warwick
+# All rights reserved.
+# ***** END GPL LICENSE BLOCK *****
+
+"""Methods of masking tissue and background."""
+
+from abc import ABC, abstractmethod
+
+import cv2
+import numpy as np
+from skimage.filters import threshold_otsu
+
+from tiatoolbox.utils.misc import objective_power2mpp
+
+
+class TissueMasker(ABC):
+    """Base class for tissue maskers.
+
+    Takes an image as in put and outputs a mask.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.fitted = False
+
+    @abstractmethod
+    def fit(self, images: np.ndarray, masks=None) -> None:
+        """Fit the masker to the images and parameters.
+
+        Args:
+            images (:class:`numpy.ndarray`):
+                List of images, usually WSI thumbnails. Expected shape is
+                NHWC (number images, height, width, channels).
+            mask (:class:`numpy.ndarray`):
+                Target/ground-truth masks. Expected shape is NHW (n
+                images, height, width).
+        """
+
+    @abstractmethod
+    def transform(self, images: np.ndarray) -> np.ndarray:
+        """Create and return a tissue mask.
+
+        Args:
+            thumbnail (:class:`numpy.ndarray`): RGB image, usually a WSI
+                thumbnail.
+
+        Returns:
+            np.ndarary: Map of semantic classes spatially over the WSI
+                e.g. regions of tissue vs background.
+        """
+        if not self.fitted:
+            raise Exception("Fit must be called before transform.")
+
+    def fit_transform(self, images: np.ndarray, **kwargs) -> np.ndarray:
+        """Perform :func:`fit` then :func:`transform`.
+
+        Sometimes it can be more optimal to perform both at the same
+        time for a single sample. In this case the base implementation
+        of :fun:`fit` followed by :func:`transform` can be overridden.
+
+        Args:
+            image (:class:`numpy.ndarray`): Image to create mask from.
+            kwargs (dict): Other key word arguments passed to fit.
+        """
+        self.fit(images, **kwargs)
+        return self.transform(images)
+
+
+class OtsuTissueMasker(TissueMasker):
+    """Tissue masker which uses Otsu's method to determine background.
+
+    Otsu's method
+
+    Examples:
+        >>> from tiatoolbox.tools.tissuemask import OtsuTissueMasker
+        >>> masker = OtsuTissueMasker()
+        >>> masker.fit(thumbnail)
+        >>> masks = masker.transform([thumbnail])
+
+        >>> from tiatoolbox.tools.tissuemask import OtsuTissueMasker
+        >>> masker = OtsuTissueMasker()
+        >>> masks = masker.fit_transform([thumbnail])
+    """
```

Comment:
Please add a new line at the end.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/5b0e0ecc1fcb9ce989d907483bef9a2b0b2c7aa5